tests/mock/agents.py

```diff
@@ -0,0 +1,223 @@
+
+    def assert_contract_calls(self, calls: Tuple[str]) -> None:
+        for call_name in calls:
+            assert call_name in self._SPY_CALLS, f'"{call_name}" was not called'
+
+
+class MockNucypherToken(MockContractAgent, NucypherTokenAgent):
+    """Look at me im a token!"""
+
+
+class MockStakingAgent(MockContractAgent, StakingEscrowAgent):
+    """dont forget the eggs!"""
+
+    CALLS = ('get_completed_work', )
+
+
+class MockPolicyManagerAgent(MockContractAgent, PolicyManagerAgent):
+    """The best ethereum policy manager ever"""
+
+
+class MockWorkLockAgent(MockContractAgent, WorkLockAgent):
+
+    CALLS = ('check_claim',
+             'eth_to_tokens',
+             'get_deposited_eth',
+             'get_eth_supply',
+             'get_base_deposit_rate',
+             'get_bonus_lot_value',
+             'get_bonus_deposit_rate',
+             'get_bonus_refund_rate',
+             'get_base_refund_rate',
+             'get_completed_work',
+             'get_refunded_work')
+
+    TRANSACTIONS = ('bid',
+                    'cancel_bid',
+                    'force_refund',
+                    'verify_bidding_correctness',
+                    'claim',
+                    'refund',
+                    'withdraw_compensation')
+
+    def __init__(self):
+
+        # Allow for mocking
+        economics = EconomicsFactory.get_economics(registry=Mock())
+
+        self.ATTRS = {'boosting_refund': economics.worklock_boosting_refund_rate,
+                      'slowing_refund': 1,  # TODO: another way to get this value?
+                      'start_bidding_date': economics.bidding_start_date,
+                      'end_bidding_date': economics.bidding_end_date,
+                      'end_cancellation_date': economics.cancellation_end_date,
+                      'minimum_allowed_bid': economics.worklock_min_allowed_bid,
+                      'lot_value': economics.worklock_supply}
+
+        super().__init__()
+
+
+class MockContractAgency:
+
+    # Test doubles
+    DOUBLE_AGENTS = {NucypherTokenAgent: MockNucypherToken,
+                     StakingEscrowAgent: MockStakingAgent,
+                     PolicyManagerAgent: MockPolicyManagerAgent,
+                     WorkLockAgent: MockWorkLockAgent}
+
+    class NoMockFound(ValueError):
+        """Well we hadn't made one yet"""
+
+    @classmethod
+    def get_agent(cls, agent_class, *args, **kwargs) -> MockContractAgent:
+        try:
+            double = cls.DOUBLE_AGENTS[agent_class]
+        except KeyError:
+            raise ValueError(f'No mock class available for "{str(agent_class)}"')
+        else:
+            return double()
+
+    @classmethod
+    def get_agent_by_contract_name(cls, contract_name: str, *args, **kwargs) -> MockContractAgent:
+        for agent, test_double, in cls.DOUBLE_AGENTS:
+            if test_double.registry_contract_name == contract_name:
+                return test_double
```

Comment:
```suggestion
        for agent, test_double in cls.DOUBLE_AGENTS.items():
            if test_double.registry_contract_name == contract_name:
                return test_double()
```
This makes me think that this mocked function is perhaps not used?

Reply:
:heavy_check_mark: